poet/sawtooth_poet/validator_registry/processor/handler.py

```diff
@@ -0,0 +1,333 @@
+        validator_state = ValidatorMap()
+    else:
+        validator_state = ValidatorInfo()
+        address = _get_address(validator_id)
+    try:
+        entries_list = state.get([address], timeout=STATE_TIMEOUT_SEC)
+    except FutureTimeoutError:
+        LOGGER.warning('Timeout occured on state.get([%s])', address)
+        raise InternalError('Unable to get {}'.format(address))
+
+    if len(entries_list) != 0:
+        validator_state.ParseFromString(entries_list[0].data)
+
+    return validator_state
+
+
+def _update_validator_state(state, validator_id, anti_sybil_id):
+    validator_map = _get_validator_state(state)
+    for entry in validator_map.entries:
+        if anti_sybil_id == entry.key:
+            # The validator's old signup info is stale and needs to be revoked
+            validator_info = _get_validator_state(state, validator_id)
+
+            validator_info.registered = "revoked"
+            validator_info = validator_info.SerializeToString()
+            _set_validator_info(state, validator_id, validator_info)
+        return
+
+    # This is a new validator that needs to be added to the map
+    validator_map.entries.add(key=anti_sybil_id, value=validator_id)
+    address = _get_address('validator_map')
+    try:
+        addresses = list(state.set(
+            [StateEntry(address=address,
+                        data=validator_map.SerializeToString())],
+            timeout=STATE_TIMEOUT_SEC)
+        )
+
+    except FutureTimeoutError:
+        LOGGER.warning(
+            'Timeout occured on state.set([%s, <value>])', address)
+        raise InternalError(
+            'Unable to add {} to valdiator_list'.format(validator_id))
+
+    if len(addresses) != 1:
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to add {} to validator_map'.format(validator_id))
+
+    LOGGER.info("Validator id %s was added to the validator_map",
+                validator_id)
+
+
+def _set_validator_info(state, validator_id, validator_info):
+    address = _get_address(validator_id)
+    try:
+        addresses = list(state.set(
+            [StateEntry(address=address,
+                        data=validator_info)],
+            timeout=STATE_TIMEOUT_SEC)
+        )
+    except FutureTimeoutError:
+        LOGGER.warning(
+            'Timeout occured on state.set([%s, <value>])', address)
+        raise InternalError(
+            'Unable to set validator info for {}'.format(validator_id))
+
+    if len(addresses) != 1:
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to save validator info for {}'.format(validator_id))
+
+    LOGGER.info("Validator info was set for %s", validator_id)
+
+
+class ValidatorRegistryTransactionHandler(object):
+
+    __REPORT_PRIVATE_KEY_WIF = \
+        '5Jz5Kaiy3kCiHE537uXcQnJuiNJshf2bZZn43CrALMGoCd3zRuo'
```

Comment:
This will only ever work with the simulator.  This private key is tied to simulated PoET1 enclave.  The real IAS will use different key to sign the report.